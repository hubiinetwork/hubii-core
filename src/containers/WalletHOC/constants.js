/*
 *
 * WalletHoc constants
 *
 */

export const SET_CURRENT_WALLET = 'src/WalletHOC/SET_CURRENT_WALLET';

export const INIT_WALLETS_BALANCES = 'src/WalletHOC/INIT_WALLETS_BALANCES';

export const LOAD_WALLET_BALANCES = 'src/WalletHOC/LOAD_WALLET_BALANCES';
export const LOAD_WALLET_BALANCES_SUCCESS = 'src/WalletHOC/LOAD_WALLET_BALANCES_SUCCESS';
export const LOAD_WALLET_BALANCES_ERROR = 'src/WalletHOC/LOAD_WALLET_BALANCES_ERROR';

export const LOAD_SUPPORTED_TOKENS = 'src/WalletHOC/LOAD_SUPPORTED_TOKENS';
export const LOAD_SUPPORTED_TOKENS_SUCCESS = 'src/WalletHOC/LOAD_SUPPORTED_TOKENS_SUCCESS';
export const LOAD_SUPPORTED_TOKENS_ERROR = 'src/WalletHOC/LOAD_SUPPORTED_TOKENS_ERROR';

export const LOAD_PRICES = 'src/WalletHOC/LOAD_PRICES';
export const LOAD_PRICES_SUCCESS = 'src/WalletHOC/LOAD_PRICES_SUCCESS';
export const LOAD_PRICES_ERROR = 'src/WalletHOC/LOAD_PRICES_ERROR';

export const CREATE_WALLET_FROM_MNEMONIC = 'src/WalletHOC/CREATE_WALLET_FROM_MNEMONIC';
export const CREATE_WALLET_FROM_PRIVATE_KEY = 'src/WalletHOC/CREATE_WALLET_FROM_PRIVATE_KEY';
export const CREATE_WALLET_SUCCESS = 'src/WalletHOC/CREATE_WALLET_SUCCESS';
export const CREATE_WALLET_FAILURE = 'src/WalletHOC/CREATE_WALLET_FAILURE';

export const ADD_NEW_WALLET = 'src/WalletHOC/ADD_NEW_WALLET';

export const DELETE_WALLET = 'src/WalletHOC/DELETE_WALLET';

export const DECRYPT_WALLET = 'src/WalletHOC/DECRYPT_WALLET';
export const DECRYPT_WALLET_SUCCESS = 'src/WalletHOC/DECRYPT_WALLET_SUCCESS';
export const DECRYPT_WALLET_FAILURE = 'src/WalletHOC/DECRYPT_WALLET_FAILURE';

export const SHOW_DECRYPT_WALLET_MODAL = 'src/WalletHOC/SHOW_DECRYPT_WALLET_MODAL';
export const HIDE_DECRYPT_WALLET_MODAL = 'src/WalletHOC/HIDE_DECRYPT_WALLET_MODAL';

export const TRANSACTION_CONFIRMED = 'src/WalletHOC/TRANSACTION_CONFIRMED';

export const TRANSFER = 'src/WalletHOC/TRANSFER';
export const TRANSFER_ETHER = 'src/WalletHOC/TRANSFER_ETHER';
export const TRANSFER_ERC20 = 'src/WalletHOC/TRANSFER_ERC20';
export const TRANSFER_SUCCESS = 'src/WalletHOC/TRANSFER_SUCCESS';
export const TRANSFER_ERROR = 'src/WalletHOC/TRANSFER_ERROR';

// Ledger Nano S constants
export const LEDGER_CONNECTED = 'src/WalletHOC/LEDGER_CONNECTED';
export const LEDGER_DISCONNECTED = 'src/WalletHOC/LEDGER_DISCONNECTED';
export const LEDGER_ETH_CONNECTED = 'src/WalletHOC/LEDGER_ETH_CONNECTED';
export const LEDGER_ETH_DISCONNECTED = 'src/WalletHOC/LEDGER_ETH_DISCONNECTED';
export const FETCH_LEDGER_ADDRESSES = 'src/WalletHOC/FETCH_LEDGER_ADDRESSES';
export const FETCHED_LEDGER_ADDRESS = 'src/WalletHOC/FETCHED_LEDGER_ADDRESS';
export const SAVE_LEDGER_ADDRESS = 'src/WalletHOC/SAVE_LEDGER_ADDRESS';
export const INIT_LEDGER = 'src/WalletHOC/INIT_LEDGER';
export const LEDGER_ERROR = 'src/WalletHOC/LEDGER_ERROR';
