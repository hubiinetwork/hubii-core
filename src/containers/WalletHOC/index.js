import * as React from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import PropTypes from 'prop-types';

import { Modal } from 'components/ui/Modal';
import { FormItem, FormItemLabel } from 'components/ui/Form';

import Input from 'components/ui/Input';
import Button from 'components/ui/Button';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import reducer from './reducer';
import saga from './saga';
import {
  makeSelectCurrentWallet,
  makeSelectCurrentWalletWithInfo,
  makeSelectLoading,
} from './selectors';
import {
  decryptWallet,
  hideDecryptWalletModal,
  initLedger,
  initWalletsBalances,
} from './actions';

import {
  StyledSpin,
} from './WalletHOC.style';

export default function WalletHOC(Component) {
  const HOC = getComponentHOC(Component);

  const mapStateToProps = createStructuredSelector({
    currentWallet: makeSelectCurrentWallet(),
    currentWalletWithInfo: makeSelectCurrentWalletWithInfo(),
    loading: makeSelectLoading(),
  });

  const withConnect = connect(mapStateToProps, mapDispatchToProps);

  const withReducer = injectReducer({ key: 'walletHoc', reducer });
  const withSaga = injectSaga({ key: 'walletHoc', saga });

  return compose(
    withReducer,
    withSaga,
    withConnect,
  )(HOC);
}

export function getComponentHOC(Component) {
  class HOC extends React.Component {
    constructor(...args) {
      super(...args);
      this.state = { password: null };
      this.onPasswordChange = this.onPasswordChange.bind(this);
      this.decryptWallet = this.decryptWallet.bind(this);
      this.handleKeyPress = this.handleKeyPress.bind(this);
    }

    componentDidMount() {
      this.props.initLedger();
      this.props.initWalletsBalances();
    }

    componentWillReceiveProps(nextProps) {
      if (nextProps.currentWallet.toJS().showDecryptModal && !this.props.currentWallet.toJS().showDecryptModal) {
        this.setState({ password: null });
      }
    }

    onPasswordChange(e) {
      this.setState({ password: e.target.value });
    }

    handleKeyPress(event) {
      if (event.key === 'Enter') {
        return this.decryptWallet();
      }
      return null;
    }

    decryptWallet() {
      const { currentWalletWithInfo } = this.props;
      this.props.decryptWallet(currentWalletWithInfo.get('address'), currentWalletWithInfo.get('encrypted'), this.state.password);
    }

    render() {
      const loading = this.props.loading.get('decryptingWallet');

      return (
        <div>
          <Component {...this.props} />
          <Modal
            footer={null}
            width={'585px'}
            maskClosable
            maskStyle={{ background: 'rgba(232,237,239,.65)' }}
            style={{ marginTop: '20px' }}
            visible={this.props.currentWallet.get('showDecryptModal')}
            onCancel={this.props.hideDecryptWalletModal}
            destroyOnClose
          >
            <FormItem
              label={<FormItemLabel>Please enter wallet password to proceed</FormItemLabel>}
              colon={false}
            >
              <Input value={this.state.password} onChange={this.onPasswordChange} type="password" onKeyPress={(e) => this.handleKeyPress(e)} />
            </FormItem>

            {loading ? (
              <StyledSpin
                delay={0}
                tip="Decrypting Wallet..."
              />
                ) : (
                  <Button type="primary" onClick={this.decryptWallet} disabled={!this.state.password}>
                    Confirm
                  </Button>
                )}
          </Modal>
        </div>
      );
    }
  }
  HOC.propTypes = {
    currentWallet: PropTypes.object.isRequired,
    currentWalletWithInfo: PropTypes.object.isRequired,
    initLedger: PropTypes.func.isRequired,
    initWalletsBalances: PropTypes.func.isRequired,
    decryptWallet: PropTypes.func.isRequired,
    hideDecryptWalletModal: PropTypes.func.isRequired,
    loading: PropTypes.object,
  };
  return HOC;
}

export function mapDispatchToProps(dispatch) {
  return {
    initLedger: () => dispatch(initLedger()),
    hideDecryptWalletModal: () => dispatch(hideDecryptWalletModal()),
    decryptWallet: (...args) => dispatch(decryptWallet(...args)),
    initWalletsBalances: (...args) => dispatch(initWalletsBalances(...args)),
  };
}

