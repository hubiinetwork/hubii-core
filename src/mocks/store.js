import { fromJS } from 'immutable';

import {
  walletHocMock,
} from 'containers/WalletHoc/tests/mocks/selectors';
import {
  hubiiApiHocMock,
} from 'containers/HubiiApiHoc/tests/mocks/selectors';
import {
  nahmiiHocMock,
} from 'containers/NahmiiHoc/tests/mocks/selectors';
import {
  appMock,
} from 'containers/App/tests/mocks/selectors';
import {
  ethOperationsHocMock,
} from 'containers/EthOperationsHoc/tests/mocks/selectors';

export const storeMock = fromJS({
  app: appMock,
  nahmiiHoc: nahmiiHocMock,
  hubiiApiHoc: hubiiApiHocMock,
  walletHoc: walletHocMock,
  ethOperationsHoc: ethOperationsHocMock,
});
