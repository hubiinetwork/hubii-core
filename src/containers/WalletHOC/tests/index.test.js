import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';
import WalletHOC, { getComponentHOC, mapDispatchToProps } from '../index';
import { walletsWithInfoMock, currentWalletMock } from './mocks';

describe('WalletHOC', () => {
  const initWalletsBalancesSpy = jest.fn();
  const initLedgerSpy = jest.fn();
  const props = {
    currentWallet: currentWalletMock,
    currentWalletWithInfo: walletsWithInfoMock.get(0),
    initLedger: initLedgerSpy,
    initWalletsBalances: initWalletsBalancesSpy,
    decryptWallet: () => {},
    hideDecryptWalletModal: () => {},
    loading: fromJS({
      decryptingWallet: false,
    }),
  };
  let dom;
  afterEach(() => {
    jest.clearAllMocks();
  });
  describe('shallow mount', () => {
    describe('#WalletHOC', () => {
      it('should return composed component', () => {
        const hoc = WalletHOC('div');
        expect(hoc.WrappedComponent).toBeDefined();
      });
    });
    describe('#Hoc rendering', () => {
      it('renders correctly when loading is false', () => {
        const Hoc = getComponentHOC('div');
        dom = shallow(
          <Hoc
            {...props}
          />
        );
        expect(dom).toMatchSnapshot();
      });
      it('renders correctly when loading is true', () => {
        const Hoc = getComponentHOC('div');
        dom = shallow(
          <Hoc
            {...props}
            loading={fromJS({ decryptingWallet: true })}
          />
        );
        expect(dom).toMatchSnapshot();
      });
    });
    describe('#componentWillReceiveProps', () => {
      it('should set the password state to null if prev modal display was false and new is true', () => {
        const Hoc = getComponentHOC('div');
        dom = shallow(
          <Hoc
            {...props}
          />
        );
        const nextProps = {
          currentWallet: fromJS({
            showDecryptModal: true,
          }),
        };
        const instance = dom.instance();
        instance.setState({ password: 'password' });
        instance.componentWillReceiveProps(nextProps);
        const expectedPassword = null;
        expect(instance.state.password).toEqual(expectedPassword);
      });
      it('should not set the password state to null if prev modal display was false and new is false', () => {
        const Hoc = getComponentHOC('div');
        dom = shallow(
          <Hoc
            {...props}
          />
        );
        const nextProps = {
          currentWallet: fromJS({
            showDecryptModal: false,
          }),
        };
        const instance = dom.instance();
        instance.setState({ password: 'password' });
        instance.componentWillReceiveProps(nextProps);
        const expectedPassword = 'password';
        expect(instance.state.password).toEqual(expectedPassword);
      });
    });
    describe('#componentDidMount', () => {
      it('should call initWalletsBalances prop when called', () => {
        const Hoc = getComponentHOC('div');
        dom = shallow(
          <Hoc
            {...props}
          />
        );
        const instance = dom.instance();
        instance.componentDidMount();
        expect(initWalletsBalancesSpy).toBeCalled();
      });
    });
    describe('#componentDidMount', () => {
      it('should call initLedger', () => {
        const Hoc = getComponentHOC('div');
        dom = shallow(
          <Hoc
            {...props}
          />
        );
        const instance = dom.instance();
        instance.componentDidMount();
        expect(initLedgerSpy).toBeCalled();
      });
    });
    describe('#onPasswordChange', () => {
      it('should update password to temporary state', () => {
        const Hoc = getComponentHOC('div');
        dom = shallow(
          <Hoc
            {...props}
          />
        );
        const instance = dom.instance();
        const password = 'test';
        instance.onPasswordChange({ target: { value: password } });
        expect(instance.state.password).toEqual(password);
      });
    });
    describe('#handleKeyPress', () => {
      it('should run the decryptWallet when "enter" key is pressed', () => {
        const Hoc = getComponentHOC('div');
        dom = shallow(
          <Hoc
            {...props}
          />
        );
        const event = {
          key: 'Enter',
        };
        const instance = dom.instance();
        const spy = jest.spyOn(instance, 'decryptWallet');
        instance.handleKeyPress(event);
        expect(spy).toHaveBeenCalledTimes(1);
      });
      it('should not run the decryptWallet when "enter" key is pressed', () => {
        const Hoc = getComponentHOC('div');
        dom = shallow(
          <Hoc
            {...props}
          />
        );
        const event = {
          key: '',
        };
        const instance = dom.instance();
        const spy = jest.spyOn(instance, 'decryptWallet');
        instance.handleKeyPress(event);
        expect(spy).toHaveBeenCalledTimes(0);
      });
    });
    describe('#decryptWallet', () => {
      it('should trigger decryptWallet action', () => {
        const decryptWalletSpy = jest.fn();
        const currentWalletWithInfo = fromJS({
          address: '0x00',
          encrypted: {},
        });
        const password = '123';
        const Hoc = getComponentHOC('div');
        dom = shallow(
          <Hoc
            {...props}
            decryptWallet={decryptWalletSpy}
            currentWalletWithInfo={currentWalletWithInfo}
          />
        );
        const instance = dom.instance();
        instance.setState({ password });
        instance.decryptWallet();
        expect(decryptWalletSpy).toBeCalledWith(currentWalletWithInfo.get('address'), currentWalletWithInfo.get('encrypted'), password);
      });
    });
    describe('#mapDispatchToProps', () => {
      it('should dispatch', () => {
        const dispatchSpy = jest.fn();
        const actions = mapDispatchToProps(dispatchSpy);
        Object.keys(actions).forEach((action, index) => {
          actions[action]();
          expect(dispatchSpy).toHaveBeenCalledTimes(index + 1);
        });
      });
    });
  });
});
