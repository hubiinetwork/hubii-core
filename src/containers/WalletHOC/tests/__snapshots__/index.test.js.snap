// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletHOC shallow mount #Hoc rendering renders correctly when loading is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HOC
    currentWallet={
        Immutable.Map {
            showDecryptModal: false,
        }
    }
    currentWalletDetails={Object {}}
    decryptWallet={[Function]}
    hideDecryptWalletModal={[Function]}
    loadWalletsBalances={[Function]}
    loading={
        Immutable.Map {
            decryptingWallet: false,
        }
    }
    startLedgerSync={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          currentWallet={
                    Immutable.Map {
                              showDecryptModal: false,
                    }
          }
          currentWalletDetails={Object {}}
          decryptWallet={[Function]}
          hideDecryptWalletModal={[Function]}
          loadWalletsBalances={[Function]}
          loading={
                    Immutable.Map {
                              decryptingWallet: false,
                    }
          }
          startLedgerSync={[Function]}
/>,
        <Modal.style
          destroyOnClose={true}
          footer={null}
          maskClosable={true}
          maskStyle={
                    Object {
                              "background": "rgba(232,237,239,.65)",
                            }
          }
          onCancel={[Function]}
          style={
                    Object {
                              "marginTop": "20px",
                            }
          }
          visible={false}
          width="585px"
>
          <Form.style__FormItem
                    colon={false}
                    label={
                              <Form.style__FormItemLabel>
                                        Please enter wallet password to proceed
                              </Form.style__FormItemLabel>
                    }
          >
                    <Input.style
                              onChange={[Function]}
                              onKeyPress={[Function]}
                              type="password"
                              value={null}
                    />
          </Form.style__FormItem>
          <Button.style
                    disabled={true}
                    onClick={[Function]}
                    type="primary"
          >
                    Confirm
          </Button.style>
</Modal.style>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "currentWallet": Immutable.Map {
            showDecryptModal: false,
},
          "currentWalletDetails": Object {},
          "decryptWallet": [Function],
          "hideDecryptWalletModal": [Function],
          "loadWalletsBalances": [Function],
          "loading": Immutable.Map {
            decryptingWallet: false,
},
          "startLedgerSync": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Form.style__FormItem
              colon={false}
              label={
                            <Form.style__FormItemLabel>
                                          Please enter wallet password to proceed
                            </Form.style__FormItemLabel>
              }
>
              <Input.style
                            onChange={[Function]}
                            onKeyPress={[Function]}
                            type="password"
                            value={null}
              />
</Form.style__FormItem>,
            <Button.style
              disabled={true}
              onClick={[Function]}
              type="primary"
>
              Confirm
</Button.style>,
          ],
          "destroyOnClose": true,
          "footer": null,
          "maskClosable": true,
          "maskStyle": Object {
            "background": "rgba(232,237,239,.65)",
          },
          "onCancel": [Function],
          "style": Object {
            "marginTop": "20px",
          },
          "visible": false,
          "width": "585px",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Input.style
                onChange={[Function]}
                onKeyPress={[Function]}
                type="password"
                value={null}
/>,
              "colon": false,
              "label": <Form.style__FormItemLabel>
                Please enter wallet password to proceed
</Form.style__FormItemLabel>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onChange": [Function],
                "onKeyPress": [Function],
                "type": "password",
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Confirm",
              "disabled": true,
              "onClick": [Function],
              "type": "primary",
            },
            "ref": null,
            "rendered": "Confirm",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            currentWallet={
                        Immutable.Map {
                                    showDecryptModal: false,
                        }
            }
            currentWalletDetails={Object {}}
            decryptWallet={[Function]}
            hideDecryptWalletModal={[Function]}
            loadWalletsBalances={[Function]}
            loading={
                        Immutable.Map {
                                    decryptingWallet: false,
                        }
            }
            startLedgerSync={[Function]}
/>,
          <Modal.style
            destroyOnClose={true}
            footer={null}
            maskClosable={true}
            maskStyle={
                        Object {
                                    "background": "rgba(232,237,239,.65)",
                                  }
            }
            onCancel={[Function]}
            style={
                        Object {
                                    "marginTop": "20px",
                                  }
            }
            visible={false}
            width="585px"
>
            <Form.style__FormItem
                        colon={false}
                        label={
                                    <Form.style__FormItemLabel>
                                                Please enter wallet password to proceed
                                    </Form.style__FormItemLabel>
                        }
            >
                        <Input.style
                                    onChange={[Function]}
                                    onKeyPress={[Function]}
                                    type="password"
                                    value={null}
                        />
            </Form.style__FormItem>
            <Button.style
                        disabled={true}
                        onClick={[Function]}
                        type="primary"
            >
                        Confirm
            </Button.style>
</Modal.style>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "currentWallet": Immutable.Map {
              showDecryptModal: false,
},
            "currentWalletDetails": Object {},
            "decryptWallet": [Function],
            "hideDecryptWalletModal": [Function],
            "loadWalletsBalances": [Function],
            "loading": Immutable.Map {
              decryptingWallet: false,
},
            "startLedgerSync": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Form.style__FormItem
                colon={false}
                label={
                                <Form.style__FormItemLabel>
                                                Please enter wallet password to proceed
                                </Form.style__FormItemLabel>
                }
>
                <Input.style
                                onChange={[Function]}
                                onKeyPress={[Function]}
                                type="password"
                                value={null}
                />
</Form.style__FormItem>,
              <Button.style
                disabled={true}
                onClick={[Function]}
                type="primary"
>
                Confirm
</Button.style>,
            ],
            "destroyOnClose": true,
            "footer": null,
            "maskClosable": true,
            "maskStyle": Object {
              "background": "rgba(232,237,239,.65)",
            },
            "onCancel": [Function],
            "style": Object {
              "marginTop": "20px",
            },
            "visible": false,
            "width": "585px",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Input.style
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  type="password"
                  value={null}
/>,
                "colon": false,
                "label": <Form.style__FormItemLabel>
                  Please enter wallet password to proceed
</Form.style__FormItemLabel>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onChange": [Function],
                  "onKeyPress": [Function],
                  "type": "password",
                  "value": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Confirm",
                "disabled": true,
                "onClick": [Function],
                "type": "primary",
              },
              "ref": null,
              "rendered": "Confirm",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`WalletHOC shallow mount #Hoc rendering renders correctly when loading is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HOC
    currentWallet={
        Immutable.Map {
            showDecryptModal: false,
        }
    }
    currentWalletDetails={Object {}}
    decryptWallet={[Function]}
    hideDecryptWalletModal={[Function]}
    loadWalletsBalances={[Function]}
    loading={
        Immutable.Map {
            decryptingWallet: true,
        }
    }
    startLedgerSync={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          currentWallet={
                    Immutable.Map {
                              showDecryptModal: false,
                    }
          }
          currentWalletDetails={Object {}}
          decryptWallet={[Function]}
          hideDecryptWalletModal={[Function]}
          loadWalletsBalances={[Function]}
          loading={
                    Immutable.Map {
                              decryptingWallet: true,
                    }
          }
          startLedgerSync={[Function]}
/>,
        <Modal.style
          destroyOnClose={true}
          footer={null}
          maskClosable={true}
          maskStyle={
                    Object {
                              "background": "rgba(232,237,239,.65)",
                            }
          }
          onCancel={[Function]}
          style={
                    Object {
                              "marginTop": "20px",
                            }
          }
          visible={false}
          width="585px"
>
          <Form.style__FormItem
                    colon={false}
                    label={
                              <Form.style__FormItemLabel>
                                        Please enter wallet password to proceed
                              </Form.style__FormItemLabel>
                    }
          >
                    <Input.style
                              onChange={[Function]}
                              onKeyPress={[Function]}
                              type="password"
                              value={null}
                    />
          </Form.style__FormItem>
          <WalletHOC.style__StyledSpin
                    delay={0}
                    tip="Decrypting Wallet..."
          />
</Modal.style>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "currentWallet": Immutable.Map {
            showDecryptModal: false,
},
          "currentWalletDetails": Object {},
          "decryptWallet": [Function],
          "hideDecryptWalletModal": [Function],
          "loadWalletsBalances": [Function],
          "loading": Immutable.Map {
            decryptingWallet: true,
},
          "startLedgerSync": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Form.style__FormItem
              colon={false}
              label={
                            <Form.style__FormItemLabel>
                                          Please enter wallet password to proceed
                            </Form.style__FormItemLabel>
              }
>
              <Input.style
                            onChange={[Function]}
                            onKeyPress={[Function]}
                            type="password"
                            value={null}
              />
</Form.style__FormItem>,
            <WalletHOC.style__StyledSpin
              delay={0}
              tip="Decrypting Wallet..."
/>,
          ],
          "destroyOnClose": true,
          "footer": null,
          "maskClosable": true,
          "maskStyle": Object {
            "background": "rgba(232,237,239,.65)",
          },
          "onCancel": [Function],
          "style": Object {
            "marginTop": "20px",
          },
          "visible": false,
          "width": "585px",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Input.style
                onChange={[Function]}
                onKeyPress={[Function]}
                type="password"
                value={null}
/>,
              "colon": false,
              "label": <Form.style__FormItemLabel>
                Please enter wallet password to proceed
</Form.style__FormItemLabel>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onChange": [Function],
                "onKeyPress": [Function],
                "type": "password",
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "delay": 0,
              "tip": "Decrypting Wallet...",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            currentWallet={
                        Immutable.Map {
                                    showDecryptModal: false,
                        }
            }
            currentWalletDetails={Object {}}
            decryptWallet={[Function]}
            hideDecryptWalletModal={[Function]}
            loadWalletsBalances={[Function]}
            loading={
                        Immutable.Map {
                                    decryptingWallet: true,
                        }
            }
            startLedgerSync={[Function]}
/>,
          <Modal.style
            destroyOnClose={true}
            footer={null}
            maskClosable={true}
            maskStyle={
                        Object {
                                    "background": "rgba(232,237,239,.65)",
                                  }
            }
            onCancel={[Function]}
            style={
                        Object {
                                    "marginTop": "20px",
                                  }
            }
            visible={false}
            width="585px"
>
            <Form.style__FormItem
                        colon={false}
                        label={
                                    <Form.style__FormItemLabel>
                                                Please enter wallet password to proceed
                                    </Form.style__FormItemLabel>
                        }
            >
                        <Input.style
                                    onChange={[Function]}
                                    onKeyPress={[Function]}
                                    type="password"
                                    value={null}
                        />
            </Form.style__FormItem>
            <WalletHOC.style__StyledSpin
                        delay={0}
                        tip="Decrypting Wallet..."
            />
</Modal.style>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "currentWallet": Immutable.Map {
              showDecryptModal: false,
},
            "currentWalletDetails": Object {},
            "decryptWallet": [Function],
            "hideDecryptWalletModal": [Function],
            "loadWalletsBalances": [Function],
            "loading": Immutable.Map {
              decryptingWallet: true,
},
            "startLedgerSync": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Form.style__FormItem
                colon={false}
                label={
                                <Form.style__FormItemLabel>
                                                Please enter wallet password to proceed
                                </Form.style__FormItemLabel>
                }
>
                <Input.style
                                onChange={[Function]}
                                onKeyPress={[Function]}
                                type="password"
                                value={null}
                />
</Form.style__FormItem>,
              <WalletHOC.style__StyledSpin
                delay={0}
                tip="Decrypting Wallet..."
/>,
            ],
            "destroyOnClose": true,
            "footer": null,
            "maskClosable": true,
            "maskStyle": Object {
              "background": "rgba(232,237,239,.65)",
            },
            "onCancel": [Function],
            "style": Object {
              "marginTop": "20px",
            },
            "visible": false,
            "width": "585px",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Input.style
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  type="password"
                  value={null}
/>,
                "colon": false,
                "label": <Form.style__FormItemLabel>
                  Please enter wallet password to proceed
</Form.style__FormItemLabel>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onChange": [Function],
                  "onKeyPress": [Function],
                  "type": "password",
                  "value": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "delay": 0,
                "tip": "Decrypting Wallet...",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
